### ISSUE TITLE
Time.parse doesn't work for 12 hour times between noon and 1pm

### ISSUE THREAD

**Issue by atlantis** [crystal-lang/crystal#9333]
I may be crazy, but this:

`puts Time.parse("2020-10-10 12:30 PM", "%Y-%m-%d %l:%M %P", Time::Location::UTC).inspect`

Throws:

```Unhandled exception: Invalid time (ArgumentError)
  from /usr/local/Cellar/crystal/0.34.0/src/time.cr:417:7 in 'local:nanosecond:location'
  from /usr/local/Cellar/crystal/0.34.0/src/time/format/parser.cr:55:16 in 'time'
  from /usr/local/Cellar/crystal/0.34.0/src/time/format.cr:84:5 in 'parse'
  from /usr/local/Cellar/crystal/0.34.0/src/time/format.cr:81:3 in 'parse'
  from /usr/local/Cellar/crystal/0.34.0/src/time.cr:1163:5 in 'parse'
```

However, other 12 hour times work just fine:

`puts Time.parse("2020-10-10 01:30 PM", "%Y-%m-%d %l:%M %P", Time::Location::UTC).inspect`

Am I missing something? The mention of `local:nanosecond:location` makes me think it might be a bug?  Thanks for an awesome language!

**Comment by atlantis** [comment]
Sorry forgot:

```Crystal 0.34.0 (2020-04-07)

LLVM: 10.0.0
Default target: x86_64-apple-macosx```

**Comment by straight-shoota** [comment]
It is indeed a bug. But it has nothing to do with nanoseconds.
The time parser just doesn't treat AM/PM correctly. I suppose none of the people who worked on that are familiar with that time notation 😆 I had to look it up, but I hope I fixed it: #9334

**Comment by atlantis** [comment]
Well that was fast - thank you! :)
